File: part 1/projects/arqsoft-22-23/SandwichManagerCore/src/main/java/com/example/SanwichManager/mappers/SandwichMapper.java
package com.example.SanwichManager.mappers;


import com.example.SanwichManager.domain.entities.Sandwich;
import com.example.SanwichManager.domain.valueObjects.SandwichDescription;
import com.example.SanwichManager.domain.valueObjects.shared.Designation;
import com.example.SanwichManager.domain.valueObjects.shared.Price;
import com.example.SanwichManager.dtos.sandwich.InputSandwichDTO;
import com.example.SanwichManager.dtos.sandwich.SandwichDTO;
import org.springframework.stereotype.Component;

@Component
public class SandwichMapper {

    public SandwichDTO tosandwichDTO(Sandwich sandwich) {
        return new SandwichDTO(sandwich.getSandwichId(), sandwich.getSandwichPrice().getPrice(), sandwich.getSandwichDesignation().getDesignation(),
                sandwich.getSandwichDescription().getDescription());
    }

    public Sandwich toSandwich(InputSandwichDTO inputSandwichDTO) {
        return new Sandwich(new Price(inputSandwichDTO.sandwichPrice), new Designation(inputSandwichDTO.sandwichDesignation),
                new SandwichDescription(inputSandwichDTO.sandwichDescription));
    }

}

File: part 2/projects/microservices/Sandwich/SandwichM/Mappers/SandwichMapper.cs
using Sandwich.Entities.DTO;
using Sandwich.Entities.Entity;

namespace Sandwich.Mappers;

public class SandwichMapper
{
    public static SandwichDTO ToDTO(Sandwiches sandwich)
    {
        var id = sandwich.SandwichId.ToString();
        var price = sandwich.SandwichPrice;
        var description = sandwich.SandwichDescription;
        var designation = sandwich.SandwichDesignation;

        return new SandwichDTO(id, price, description, designation);
    }

    public static Sandwiches ToDomain(SandwichDTO sandwichDto)
    {
        var id = int.Parse(sandwichDto.SandwichId);
        var price = sandwichDto.SandwichPrice;
        var description = sandwichDto.SandwichDescription;
        var designation = sandwichDto.SandwichDesignation;
      
        return new Sandwiches(id, price, description, designation);
    }
    
    public static IQueryable<SandwichDTO> ToDTOList(IQueryable<Sandwiches> sandwiches)
    {
        List<SandwichDTO> list = new List<SandwichDTO>();
        foreach(Sandwiches s in sandwiches)
        {
            list.Add(ToDTO(s));
        }

        return list.AsQueryable();
    }


}